File: 53-maximum-subarray/solution.py
class Solution:
  def maxSubArray(self, nums: list[int]) -> int:
    local_max = result = nums[0]
    for i in range(1, len(nums)):
      num = nums[i]
      local_max += num
      if local_max < num:
        local_max = num

      result = max(result, local_max)

    return result

File: 53-maximum-subarray/solution.cpp
#include <vector>
#include <algorithm>

class Solution
{
public:
  int maxSubArray(std::vector<int> &nums)
  {
    int local_max = nums[0], result = nums[0];
    for (size_t i = 1; i < nums.size(); ++i)
    {
      const int num = nums[i];
      local_max += num;
      if (local_max < num)
      {
        local_max = num;
      }

      result = std::max(result, local_max);
    }

    return result;
  }
};